workflow-templates/im-build-and-test-generic.yml

```diff
@@ -0,0 +1,328 @@
+#    - This workflow should only be used once per repository
+#
+# Projects to use this Template with:
+#    - any repository with deployables
+#    - monorepo style repositories
+#
+# TODO: Prerequisites:
+#    - Ensure each of the repo-level and env-level secrets used in this workflow have been populated by an admin in your repository.
+
+name: 🏭 CI Build
+
+on:
+  # This workflow will run for the PR events:
+  # opened/reopened: A PR is opened/re-opened,
+  # synchronize: The PR's head branch is updated (commits pushed, the base branch changed or the head is updated from the base branch)
+  # closed: A PR is merged or closed (The examine-triggers job makes sure we don't build anything if a PR is closed without being merged)
+  pull_request:
+    types: [opened, reopened, synchronize, closed]
+
+env:
+  READ_PKG_TOKEN: ${{ secrets.READ_PKG_TOKEN }} # This is an org-level secret
+  REPO_URL: ${{ github.server_url }}/${{ github.repository }}
+  MS_TEAMS_CONNECTOR: '' # TODO: supply the Teams webhook URL
+  TIMEZONE: America/Denver # TODO: Verify timezone
+
+jobs:
+  examine-triggers:
+    runs-on: ubuntu-latest
+    outputs:
+      CONTINUE_WORKFLOW: ${{ env.CONTINUE_WORKFLOW }}
+      CREATE_RELEASE: ${{ env.CREATE_RELEASE }}
+      IS_PRERELEASE: ${{ env.IS_PRERELEASE }}
+      NEXT_VERSION: ${{ env.NEXT_VERSION }}
+      REF_TO_BUILD_AND_TAG: ${{ env.REF_TO_BUILD_AND_TAG }}
+
+    steps:
+      - name: Set default env variables
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const targetRef = '${{ github.base_ref }}';
+            const sourceRef = '${{ github.head_ref }}';
+            const mergeRef = '${{ github.ref }}';
+
+            const prIsDraft = '${{ github.event.pull_request.draft }}' === 'true';
+            const prClosed = '${{ github.event.action }}' === 'closed';
+            const prMerged = prClosed && '${{ github.event.pull_request.merged }}' === 'true';
+            const prMergedToMain = prMerged && targetRef === 'main';
+            const isPreRelease = !prMergedToMain;
+
+            // For a detailed explanation of why we use different refs for different scenarios 
+            // see https://docs.github.com/en/rest/reference/pulls#get-a-pull-request
+            const refToBuildAndTag = prMergedToMain ? mergeRef : sourceRef;
+
+            const continueWorkflow = prClosed && !prMerged ? false : true;
+            const doTagRelease = continueWorkflow && !prIsDraft ? true :  false;
+
+            core.exportVariable('IS_PRERELEASE', isPreRelease);
+            core.exportVariable('REF_TO_BUILD_AND_TAG', refToBuildAndTag);
+            core.exportVariable('CONTINUE_WORKFLOW', continueWorkflow);
+            core.exportVariable('CREATE_RELEASE', doTagRelease);
+
+      # NOTE: This workflow uses one version tag with no prefix for all the deployables in the repository.
+      #       If you want each deployable to have it's own version tag prefix, this workflow is not for you.
```

Comment:
Do you think it makes sense to put this higher up so it's easier to see? And would it make sense to point people to some other template they can use if they have that use case? Or maybe give instruction on how to update this workflow, something along the lines of [this](https://github.com/im-practices/.github/blob/main/workflow-templates/im-build-dotnet-ci.yml#L375)?

Also, I think we've trained people to search for TODO's when implementing these templates, but not NOTE's, so it might make sense to make this a TODO if we leave it here. If we move it up to the top, it could probably go along with section that specifies what projects to use this template for.